internal/handler/userHandler_test.go

```diff
@@ -0,0 +1,290 @@
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.UserResponseDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusCreated, recorder.Code)
+	assert.Equal(t, createdUser.GetID(), responseBody.ID)
+	assert.Equal(t, createdUser.GetFirstName(), responseBody.FirstName)
+	assert.Equal(t, createdUser.GetLastName(), responseBody.LastName)
+	assert.Equal(t, createdUser.GetEmail(), responseBody.Email)
+	assert.Equal(t, createdUser.GetAge(), responseBody.Age)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+
+func TestErrorPersistingUser(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "John",
+		LastName:  "Doe",
+		Email:     "johndoe@email.com",
+		Age:       18,
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	user, _ := mapper.UserRequestToUser(requestBody)
+	mockService.EXPECT().SaveUser(ctx, user).Return(nil, errors.New("some error")).Once()
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+	assert.Equal(t, http.StatusInternalServerError, recorder.Code)
+	assert.Equal(t, "Something went wrong", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.Nil(t, responseBody.Field)
+	mockService.AssertExpectations(t)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
+func TestBindJsonFailure(t *testing.T) {
+	gin.SetMode(gin.TestMode)
+	recorder := httptest.NewRecorder()
+	ctx, _ := gin.CreateTestContext(recorder)
+	mockService := mocks.NewService(t)
+
+	requestBody := request.UserRequestDto{
+		FirstName: "",
+		LastName:  "",
+		Email:     "",
+		Age:       17,
+	}
+
+	jsonBody, _ := json.Marshal(requestBody)
+	ctx.Request = &http.Request{
+		Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+	}
+
+	handler := &Handler{
+		userService: mockService,
+	}
+
+	handler.Save(ctx)
+	var responseBody response.ErrorDto
+	_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+	assert.Equal(t, http.StatusBadRequest, recorder.Code)
+	assert.Equal(t, "Error while binding JSON", responseBody.Message)
+	assert.NotEmpty(t, responseBody.Timestamp)
+	assert.NotEmpty(t, responseBody.Field)
+
+	fields, _ := convertToMap(responseBody.Field)
+	assert.Equal(t, "required", fields["firstName"])
+	assert.Equal(t, "required", fields["lastName"])
+	assert.Equal(t, "required", fields["email"])
+	assert.Equal(t, "age must be greater than 18", fields["age"])
+	mockService.AssertNumberOfCalls(t, "SaveUser", 0)
+	mockService.AssertNumberOfCalls(t, "GetUserById", 0)
+}
```

Comment:
These lines are not required, since no .EXPECT() was called any method calls on the mock will cause an error.

Reply:
Fixed